packages/terra-popup/src/Popup.jsx

```diff
@@ -0,0 +1,370 @@
+};
+
+const MIRROR_TB = {
+  middle: 'middle',
+  top: 'bottom',
+  bottom: 'top',
+};
+
+const BASE_WIDTH = 16;
+const BASE_HEIGHT = 9;
+
+const DIMENSIONS = [
+  '10x',
+  '25x',
+  '50x',
+  '75x',
+  '100x',
+];
+
+const DIMENSIONS_MAP = {
+  '10x': 10,
+  '25x': 25,
+  '50x': 50,
+  '75x': 75,
+  '100x': 100,
+};
+
+const COMBINE = () => {
+  const combinations = [];
+  for (let i = 0; i < DIMENSIONS.length; i += 1) {
+    for (let j = 0; j < DIMENSIONS.length; j += 1) {
+      combinations.push(`${DIMENSIONS[i]} ${DIMENSIONS[j]}`);
+    }
+  }
+  return combinations;
+};
+
+const DIMENSION_COMBINATIONS = COMBINE();
+
+const propTypes = {
+  /**
+   * Bounding container for the popup, will use window if no value provided.
+   */
+  boundingRef: PropTypes.func,
+  /**
+   * The children to be displayed as content within the popup.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * CSS classnames that are append to the arrow.
+   */
+  classNameArrow: PropTypes.string,
+  /**
+   * CSS classnames that are append to the popup content inner.
+   */
+  classNameContent: PropTypes.string,
+  /**
+   * CSS classnames that are append to the overlay.
+   */
+  classNameOverlay: PropTypes.string,
+  /**
+   * Attachment point for the popup, this will be mirrored to the target.
+   */
+  contentAttachment: PropTypes.oneOf(TetherComponent.attachmentPositions),
+  /**
+   * A string pair of breakpoint sizes, ('10x', '25x', '50x', '75x', '10x'), i.e. 'vertical horizontal'
+   */
+  contentDimensions: PropTypes.oneOf(DIMENSION_COMBINATIONS),
+  /**
+   * Should an arrow be placed at the attachment point.
+   */
+  isArrowDisplayed: PropTypes.bool,
+  /**
+   * Should the default behavior, that inserts a header when constraints are breached, be disabled.
+   */
+  isHeaderDisabled: PropTypes.bool,
+  /**
+   * Should the popup be presented as open.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * Target element for the popup to anchor to.
+   */
+  targetRef: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  boundingRef: null,
+  classNameArrow: null,
+  classNameContent: null,
+  classNameOverlay: null,
+  contentAttachment: 'top center',
+  contentDimensions: '25x 25x',
+  isArrowDisplayed: false,
+  isHeaderDisabled: false,
+  isOpen: false,
+};
+
+class Popup extends React.Component {
+
+  static mirrorAttachment(attachment) {
+    const parsedValue = Popup.parseStringPair(attachment);
+    let horizontal = parsedValue.horizontal;
+    let vertical = parsedValue.vertical;
+
+    if (parsedValue.vertical === 'middle') {
+      horizontal = MIRROR_LR[parsedValue.horizontal];
+    } else {
+      vertical = MIRROR_TB[parsedValue.vertical];
+    }
+
+    return `${vertical} ${horizontal}`;
+  }
+
+  static getContentOffset(attachment, targetNode, arrowOffset) {
```

Comment:
I think documentation of these methods that are doing the positions and offset logic is needed.